testutil/daemon/daemon.go

```diff
@@ -262,6 +262,7 @@ func (d *Daemon) LogFileName() string {
 
 // ReadLogFile returns the content of the daemon log file
 func (d *Daemon) ReadLogFile() ([]byte, error) {
+	_ = d.logFile.Sync()
```

Comment:
YOLO; wondering if it's just not written to disk the moment we try to read it 🤔

Reply:
Looks "better", but still flaky

So, looks like test-integration-flaky (perhaps because of the subtests) runs the test 25 times in total:

```bash
cd /Users/sebastiaan/Downloads/bundles/1/test-integration-flaky/TestDaemonProxy/reload_sanitized

ls -l
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d03a7aefa903c/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d12825289bf2b/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d35d0bbf7367a/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d47371778e153/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d4c7283bf1c32/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d59a9befb682b/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d5c38961c3cca/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d5c6cf3878a96/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d5f09c9777209/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d6e6a3dc8339f/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d7ef34388452f/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d7fc5ed00fc65/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d83c195f1c19d/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d9ac12b0678f4/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 d9da71ee7bf96/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 db95143c8d488/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 dba5677b1543e/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 dbe1bc8fb4409/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 dc09093c6ca96/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 dc3a50433e7c6/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 dc7d0850ec28a/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 dd0482bb66e57/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 dd4312712c345/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 ddc2b2078d2bb/
drwxr-xr-x@ 3 sebastiaan  staff  96 Sep 21 16:06 def3e032e5255/
```

Checking if logs for all of those contains the `Reloaded configuration` string, that looks to be the case:

```bash
grep -lr 'Reloaded configuration' . | wc -l
25
```

But the test failed twice;

```bash
grep -r 'does not contain "Reloaded configuration:"' . | wc -l
2
```

Reply:
The failure that was reported was in `d9ac12b0678f4`, but it also has the string, but I do see it ran into another failure with the libnetwork socket, but that looks to be in the shutdown cycle;

```
time="2021-09-21T12:55:04.290097255Z" level=info msg="API listen on /tmp/docker-integration/d9ac12b0678f4.sock"
time="2021-09-21T12:55:04.414946112Z" level=debug msg="Calling GET /_ping"
time="2021-09-21T12:55:04.415317278Z" level=debug msg="Calling GET /info"
time="2021-09-21T12:55:04.430754586Z" level=info msg="Got signal to reload configuration, reloading from: /etc/docker/daemon.json"
time="2021-09-21T12:55:04.431178648Z" level=debug msg="Reset Max Concurrent Downloads: 3"
time="2021-09-21T12:55:04.431201851Z" level=debug msg="Reset Max Concurrent Uploads: 5"
time="2021-09-21T12:55:04.431221426Z" level=debug msg="Reset Max Download Attempts: 5"
time="2021-09-21T12:55:04.432309643Z" level=info msg="Reloaded configuration: {\"storage-driver\":\"overlay2\",\"mtu\":1500,\"pidfile\":\"/go/src/github.com/docker/docker/bundles/test-integration-flaky/TestDaemonProxy/reload_sanitized/d9ac12b0678f4/docker.pid\",\"data-root\":\"/go/src/github.com/docker/docker/bundles/test-integration-flaky/TestDaemonProxy/reload_sanitized/d9ac12b0678f4/root\",\"exec-root\":\"/tmp/dxr/d9ac12b0678f4\",\"group\":\"docker\",\"deprecated-key-path\":\"/etc/docker/key.json\",\"max-concurrent-downloads\":3,\"max-concurrent-uploads\":5,\"max-download-attempts\":5,\"shutdown-timeout\":15,\"debug\":true,\"hosts\":[\"unix:///tmp/docker-integration/d9ac12b0678f4.sock\"],\"log-level\":\"info\",\"swarm-default-advertise-addr\":\"\",\"swarm-raft-heartbeat-tick\":0,\"swarm-raft-election-tick\":0,\"metrics-addr\":\"\",\"host-gateway-ip\":\"172.18.0.1\",\"log-driver\":\"json-file\",\"ip\":\"0.0.0.0\",\"icc\":true,\"iptables\":true,\"ip-forward\":true,\"ip-masq\":true,\"userland-proxy\":true,\"default-address-pools\":{\"Values\":null},\"network-control-plane-mtu\":1500,\"experimental\":false,\"containerd\":\"/var/run/docker/containerd/containerd.sock\",\"builder\":{\"GC\":{},\"Entitlements\":{}},\"containerd-namespace\":\"d9ac12b0678f4\",\"containerd-plugin-namespace\":\"d9ac12b0678f4p\",\"runtimes\":{\"io.containerd.runc.v2\":{\"path\":\"runc\"},\"io.containerd.runtime.v1.linux\":{\"path\":\"runc\"},\"runc\":{\"path\":\"runc\"}},\"default-runtime\":\"runc\",\"seccomp-profile\":\"builtin\",\"default-shm-size\":67108864,\"default-ipc-mode\":\"private\",\"default-cgroupns-mode\":\"host\",\"resolv-conf\":\"/etc/resolv.conf\",\"http-proxy\":\"https://xxxxx:xxxxx@example.org\",\"https-proxy\":\"https://xxxxx:xxxxx@example.org\",\"no-proxy\":\"example.com\"}"
time="2021-09-21T12:55:04.435328719Z" level=info msg="Processing signal 'interrupt'"
time="2021-09-21T12:55:04.435439253Z" level=debug msg="daemon configured with a 15 seconds minimum shutdown timeout"
time="2021-09-21T12:55:04.435514624Z" level=debug msg="start clean shutdown of all containers with a 15 seconds timeout..."
time="2021-09-21T12:55:04.435590112Z" level=debug msg="found 0 orphan layers"
time="2021-09-21T12:55:04.437848674Z" level=debug msg="Unix socket /tmp/dxr/d9ac12b0678f4/libnetwork/f7725651065a.sock doesn't exist. cannot accept client connections"
time="2021-09-21T12:55:04.437938435Z" level=debug msg="Cleaning up old mountid : start."
time="2021-09-21T12:55:04.439961024Z" level=debug msg="Cleaning up old mountid : done."
time="2021-09-21T12:55:04.444201003Z" level=debug msg="Clean shutdown succeeded"
time="2021-09-21T12:55:04.444236880Z" level=info msg="Daemon shutdown complete"
```

The same failure is appearing in all of them, it seems:

```bash
grep -lr 'cannot accept client connections' . | wc -l
25
```

Actually; perhaps even in all tests that shutdown the daemon;

```bash
grep -lr 'cannot accept client connections' . | wc -l
     419
```